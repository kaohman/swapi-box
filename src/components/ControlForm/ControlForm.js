import React from 'react';

const ControlForm = ({ getData, activeButton, favoritesCount }) => {
  const buttonLabels = ['People', 'Planets', 'Vehicles'];
  return (
    <div id='button-container'>
    {
      buttonLabels.map(label => {
        return (
          <button 
            onClick={(event) => getData(event.target.id)}
            key={label}
            id={label.toLowerCase()}
            className= {
              activeButton === label.toLowerCase() ? 'active' : 'inactive'
            }
          >{label}</button>
        )
      })
    }
      <button 

        id='favorites'
        className={activeButton === 'favorites' ? 'active' : 'inactive'}
      >
        View {favoritesCount} Favorites
      </button>
    </div>
  );
}

export default ControlForm;