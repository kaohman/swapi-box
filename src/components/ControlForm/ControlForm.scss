#button-container {
  width: 80%;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 20px;
  margin: 10px auto;
}

button {
  border-radius: 5px;
  background-color: lightgrey;
  padding: 10px;
  border: none;

  &:hover {
    background-color: aqua;
  }

}

.active {
  background-color: purple;
}