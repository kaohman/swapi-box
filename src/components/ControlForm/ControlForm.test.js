import React from 'react';
import ReactDOM from 'react-dom';
import ControlForm from './ControlForm';
import { shallow } from 'enzyme';

describe('ControlForm', () => {
  let wrapper;
  let mockGetData;
  let mockShowFavorites;
  let mockCurrentPage;
  let mockFavoritesCount;
  let mockEvent;

  beforeEach(() => {
    mockGetData = jest.fn();
    mockShowFavorites = jest.fn();
    mockCurrentPage = 'people';
    mockFavoritesCount = 0;
    mockEvent = { target: { id: '' } };
    wrapper = shallow(
      <ControlForm
        activeButton={mockCurrentPage}
        getData={mockGetData}
        favoritesCount={mockFavoritesCount}
        showFavorites={mockShowFavorites}
      />
    );
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call getData on button click', () => {// execution
    wrapper.find('#people').simulate('click', mockEvent);
    // expectation
    expect(mockGetData).toHaveBeenCalled();
  });

  it('should call showFavorites on button click', () => {
    wrapper.find('#favorites').simulate('click', mockEvent);
    // expectation
    expect(mockShowFavorites).toHaveBeenCalled();
  });
});