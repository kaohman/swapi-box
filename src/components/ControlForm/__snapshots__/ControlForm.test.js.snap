// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlForm should match the snapshot with all data passed in 1`] = `
<div
  id="button-container"
>
  <button
    className="active"
    id="people"
    key="People"
    onClick={[Function]}
  >
    People
  </button>
  <button
    className="inactive"
    id="planets"
    key="Planets"
    onClick={[Function]}
  >
    Planets
  </button>
  <button
    className="inactive"
    id="vehicles"
    key="Vehicles"
    onClick={[Function]}
  >
    Vehicles
  </button>
  <button
    className="inactive"
    id="favorites"
    onClick={[Function]}
  >
    View 
    0
     Favorites
  </button>
</div>
`;
